torchvision/datasets/mnist.py

```diff
@@ -435,6 +435,74 @@ def get_int(b: bytes) -> int:
     return int(codecs.encode(b, 'hex'), 16)
 
 
+class MovingMNIST(VisionDataset):
+    """MovingMNIST"""
+    url = "http://www.cs.toronto.edu/~nitish/unsupervised_video/mnist_test_seq.npy"
+    file = "moving_mnist.pt"
+
+    def __init__(
+                 self,
+                 root: str, 
+                 transform: Optional[Callable] = None,
+                 target_transform: Optional[Callable] = None,
+                 download: bool = False):
+        super(MovingMNIST, self).__init__(root, transform, target_transform)
+
+        if download:
+            self.download()
+
+        if not self._check_exists():
+            raise RuntimeError('Dataset not found.' +
+                               ' You can use download=True to download it')
+
+        self.data = torch.load(os.path.join(self.processed_folder, self.file))        
+
+    def download(self) -> None:
+        if self._check_exists():
+            return
+        os.makedirs(self.raw_folder, exist_ok=True)
+        os.makedirs(self.processed_folder, exist_ok=True)
+
+        filename = self.url.rpartition('/')[2]
+        file_path = os.path.join(self.raw_folder, filename)
+        if not os.path.isfile(file_path):
+            download_url(self.url, root=self.raw_folder, filename=filename)
+        data = read_npy_file(file_path)
+
+        with open(os.path.join(self.processed_folder, self.file), 'wb') as f:
+            torch.save((data), f)
+
+    def _check_exists(self) -> bool:
+        return (os.path.exists(os.path.join(self.processed_folder,
+                                            self.file)))
+
+
+    def __getitem__(self, index: int) -> Tuple[Any, Any]:
```

Comment:
Without a length, the dataset is not iterable.

```suggestion
    def __len__(self) -> int:
        return len(self.data)
        
    def __getitem__(self, index: int) -> Tuple[Any, Any]:
```